_stbt/state_watch.py

```diff
@@ -211,6 +215,36 @@ def write(self, data):
                     "Error processing state change: %s" % str(e))
 
 
+def activate(test_script, trace_file=None):
+    """Activate the state-watch tracer."""
+    global _TRACER
+    _TRACER = new_state_sender(trace_file)  # pylint: disable=W0212
+
+    absfilepath, funcname, function = utils.parse_test_uri(test_script)
+
+    def tracefunc(frame_, event, _):
+        if event == "line" and frame_.f_code.co_filename == absfilepath:
+            _TRACER.log_current_line(frame_.f_code.co_filename, frame_.f_lineno)
+        return tracefunc
+
+    sys.settrace(tracefunc)
+    _TRACER.log_test_starting(
+        test_script, os.path.relpath(absfilepath), funcname,
+        function.func_code.co_firstlineno if function is not None else 1)
+
+
+def deactivate():
+    """Deactivate the state-watch tracer.
+
+
+    If the tracer has not been `activate`d, this function is a no-op.
+    """
+    if _TRACER is not None:
+        sys.settrace(None)
+        _TRACER.log_test_ended()
+        _TRACER.close()
```

Comment:
> Seems like a fair bit of extra work if it's simply to work around having a global variable?

I doesn't seem too difficult to me.  Either way I've been putting in a bunch of effort to remove global state (#297 #333) so I don't want to merge something that introduces more.

> It's not like you would/could instantiate multiple tracers because of the sys.settrace limitation, and ultimately it's not user-facing

[`sys.settrace` docs](https://docs.python.org/2/library/sys.html#sys.settrace) say:

> The function is thread-specific; for a debugger to support multiple threads, it must be registered using settrace() for each thread being debugged.

And we could use `sys.gettrace` to chain the trace functions together, not that I'm suggesting that this would be necessary for this pull request.

> much nicer to read (and easier to understand) than
> 
>     if _tracer is not None:
>         _tracer.__exit__(None, None, None)
￼
Note: Under my proposed scheme `_tracer` in `stbt-run` would never be `None`, it just might not install a trace handler.